File: jscomp/test/nested_obj_literal.ml
let structural_obj = [%bs.obj {x= {y= {z= 3}}}]

(* compiler inferred type : val structural_obj : < x : < y : < z : int > > >
   [@bs.obj] *)

type 'a x = {x: 'a}
type 'a y = {y: 'a}
type 'a z = {z: 'a}

let f_record = {x= {y= {z= 3}}}

(* compiler inferred type : val f_record : int z y x *)

File: jscomp/test/nested_obj_literal.js
'use strict';


var structural_obj = {
  x: {
    y: {
      z: 3
    }
  }
};

var f_record = /* record */{
  x: /* record */{
    y: /* record */{
      z: 3
    }
  }
};

exports.structural_obj = structural_obj;
exports.f_record = f_record;
/* No side effect */
